symengine/dict.cpp

```diff
@@ -51,6 +51,18 @@ inline std::ostream& print_vec_rcp(std::ostream& out, T& d)
     return out;
 }
 
+template<class T>
+inline std::ostream& print_set(std::ostream& out, T& d)
+{
+    out << "{";
+    for (auto p = d.begin(); p != d.end(); p++) {
+        if (p != d.begin()) out << ", ";
+        out << (*p).first << ": " << (*p).second;
+    }
+    out << "}";
+    return out;
+}
+
```

Comment:
Hm, the only difference that I can see so far is just `p->first` vs. `(*p).first`, which is exactly the same. Is there any other difference? If you try it, what happens? Does it compile? What does it print?